src/core/storage/compression/Compressor.h

```diff
@@ -142,10 +145,10 @@ class CompressorInternal {
 	};
 
 	std::vector<std::string> m_files;//!< path to the m_files for compress
+	FILE *m_compressed_fp = nullptr; //!< file pinter to the new created compressed file
 
-	FILE *m_compressed_fp = nullptr;                 //!< file pinter to the new created compressed file
-	std::map<unsigned char, int> m_occurrence_symbol;//!< key-value pair
-	                                                 //!< in which keys are m_symbols and values are their number of occurrences
+	long int m_occurrence_symbol[256];//!< long integer array that will contain
```

Comment:
I would strongly prefer if this was replaced by C-arrays modern version - `std::array<long, 256>`. This would allow you to make use of its niceties - `.begin()`, `.end()`, `.size()`, etc. (ch. line  87 - instead of

```
std::fill(m_occurrence_symbol, m_occurrence_symbol + 256, 0);
```

it should be

```
std::fill(m_occurrence_symbol.begin(), m_occurrence_symbol.end(), 0);
```

Reply:
Resolved